contrib/emojisum-py/src/emojisum/__init__.py

```diff
@@ -0,0 +1,101 @@
+import hashlib
+import json
+import mmap
+import os
+import re
+import sys
+from pathlib import Path
+from typing import NamedTuple, Optional
+
+from rich.console import Console
+
+SHA1SUM_PATTERN = re.compile(r"(?P<digest>\S*)\s+(?P<path>.*)")
+OPENSSL_SHA1_PATTERN = re.compile(r"SHA1\((?P<path>.*)\)=\s+(?P<digest>.*)")
+OUTPUT_PATTERNS = [SHA1SUM_PATTERN, OPENSSL_SHA1_PATTERN]
+
+
+class Sha1Result(NamedTuple):
+    hex_digest: str
+    path: str
+
+
+def load_table():
+    with (Path(__file__).parent / "emojimap.json").open() as f:
+        data = json.load(f)
+
+    return data["emojiwords"]
+
+
+def _emojify(hex_digest):
+    for i in range(0, len(hex_digest), 2):
+        yield load_table()[int(hex_digest[i : i + 1], 0x10)][0]
+
+
+def emojify(hex_digest: str):
+    return "".join(_emojify(hex_digest))
+
+
+def sha1sum(path: Path):
```

Comment:
all the example of usage, with sha1 is cool, but might should go into it's own `sum` kind of package

Reply:
You mean, separating it into a different module in the code here, or something else?